node/src/tls.rs

```diff
@@ -81,7 +81,7 @@ type SslResult<T> = Result<T, ErrorStack>;
 
 /// SHA512 hash.
 #[derive(Copy, Clone, DataSize, Deserialize, Serialize)]
-struct Sha512(#[serde(with = "big_array::BigArray")] [u8; Sha512::SIZE]);
+pub struct Sha512(#[serde(with = "big_array::BigArray")] [u8; Sha512::SIZE]);
```

Comment:
I see in your PR description that you need these for tests, but I can't see where that's actually happening.

If possible, let's remove these two `pub`s again please.

Reply:
I use it in the lazy `ALICE_NODE_ID`, which is subsequently used in the tests.

Reply:
I could instead expose a function to attempt to produce a `NodeId` from a `PublicKey`, which I think would make sense to have lying around.